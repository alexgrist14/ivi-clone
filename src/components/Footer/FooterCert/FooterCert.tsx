import Link from "next/link";
import { FC } from "react";
import styles from "./FooterCert.module.sass";

const FooterCert: FC = () => {
  return (
    <div className={styles.cert}>
      <Link className={styles.cert__link} href="https://www.ivi.ru/cert">
        Активация сертификата
      </Link>
    </div>
  );
};

export default FooterCert;
