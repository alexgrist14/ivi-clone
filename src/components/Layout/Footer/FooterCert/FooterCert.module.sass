.cert
  background-clip: text
  -webkit-background-clip: text
  background-image: linear-gradient(-45deg,#c447ff,#ea003d)
  -webkit-text-fill-color: transparent
  margin-top: 8px

  &__link
    color: transparent
    font-size: 15px
    line-height: 20px
    font-weight: 500
