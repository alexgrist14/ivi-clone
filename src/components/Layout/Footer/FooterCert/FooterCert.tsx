import Link from "next/link";
import { FC } from "react";
import { useTranslation } from "next-i18next";
import styles from "./FooterCert.module.sass";

const FooterCert: FC = () => {
  const { t } = useTranslation("footer");

  return (
    <div className={styles.cert}>
      <Link className={styles.cert__link} href="https://www.ivi.ru/cert" target="_blank">
        {t("cert")}
      </Link>
    </div>
  );
};

export default FooterCert;
