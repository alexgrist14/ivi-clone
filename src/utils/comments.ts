import { IComment } from "../types/IComment";

export const mockComment: IComment = {
  comment_id: "4d522235-7915-4122-8303-4bc79ea603a5",
  title: "About film",
  text: "My comment about the film",
  like: "36",
  film_id: "eb5eb005-5818-4cee-9f7b-0fc6c1fae2cc",
  createdAt: "2023-05-08T07:30:59.761Z",
  updatedAt: "2023-05-08T07:30:59.761Z",
  user_id: "f7b2bc15-ea49-453e-a924-c0c32b21cee0",
  parent_id: null,
  user: {
    email: "admin@gmail.com",
    profile: {
      profile_id: "51ab6af9-0739-4384-8f52-f5cee6b7ca58",
      first_name: "Victor",
      last_name: "Barinov",
    },
  },
  sub_comments: [
    {
      comment_id: "ec2d35f8-b9de-4fa6-a906-c76b9f2ddaef",
      title: "About film",
      text: "My comment about the film",
      like: "36",
      film_id: "eb5eb005-5818-4cee-9f7b-0fc6c1fae2cc",
      createdAt: "2023-05-08T07:38:01.507Z",
      updatedAt: "2023-05-08T07:38:01.507Z",
      user_id: "f7b2bc15-ea49-453e-a924-c0c32b21cee0",
      parent_id: "4d522235-7915-4122-8303-4bc79ea603a5",
      user: {
        email: "admin@gmail.com",
        profile: {
          profile_id: "51ab6af9-0739-4384-8f52-f5cee6b7ca58",
          first_name: "Victor",
          last_name: "Barinov",
        },
      },
      sub_comments: [
        {
          comment_id: "4c786377-f586-4580-ab6e-24cb0ccd407e",
          title: "About film",
          text: "My comment about the film",
          like: "36",
          film_id: "eb5eb005-5818-4cee-9f7b-0fc6c1fae2cc",
          createdAt: "2023-05-08T07:38:20.229Z",
          updatedAt: "2023-05-08T07:38:20.229Z",
          user_id: "f7b2bc15-ea49-453e-a924-c0c32b21cee0",
          parent_id: "ec2d35f8-b9de-4fa6-a906-c76b9f2ddaef",
          user: {
            email: "admin@gmail.com",
            profile: {
              profile_id: "51ab6af9-0739-4384-8f52-f5cee6b7ca58",
              first_name: "Victor",
              last_name: "Barinov",
            },
          },
          sub_comments: [],
        },
      ],
    },
  ],
};
