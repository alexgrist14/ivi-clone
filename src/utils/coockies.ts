export const getCookieByName = (cookieName: string): string | null => {
  const cookies = document.cookie.split(';');

  for (let i = 0; i < cookies.length; i++) {
    const cookie = cookies[i].trim();

    if (cookie.startsWith(`${cookieName}=`)) {
      const cookieValue = cookie.substring(cookieName.length + 1);
      return decodeURIComponent(cookieValue);
    }
  }

  return null;
};