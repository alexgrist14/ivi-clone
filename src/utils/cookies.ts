export const getCookieByName = (cookieName: string): string | null => {
  const cookies = document.cookie.split(';');

  for (let i = 0; i < cookies.length; i++) {
    const cookie = cookies[i].trim();

    if (cookie.startsWith(`${cookieName}=`)) {
      const cookieValue = cookie.substring(cookieName.length + 1);
      return decodeURIComponent(cookieValue);
    }
  }

  return null;
};

export const deleteCookiesByNames = (cookieNames: string[]):void => {
  cookieNames.forEach((cookieName) => {
    document.cookie = `${cookieName}=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;`;
  });
};