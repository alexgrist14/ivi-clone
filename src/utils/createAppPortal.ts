import { ReactNode, ReactPortal } from "react";
import { createPortal } from "react-dom";
import { isBrowser } from "./isBrowser";

const createAppPortal = (node: ReactNode): ReactPortal | null => {
  console.log(isBrowser(window));

  const documentRoot = isBrowser(window)
    ? document.getElementById("__next")
    : undefined;

  console.log(documentRoot);

  return documentRoot ? createPortal(node, documentRoot) : null;
};

export default createAppPortal;
