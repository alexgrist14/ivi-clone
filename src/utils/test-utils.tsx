import { ReactNode } from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { reducersSetup } from "/src/store";
import { configureStore } from "@reduxjs/toolkit";
import { IProviderRender } from "../types/IRender";

export const renderWithProviders = (children: ReactNode): IProviderRender => {
  const store = configureStore({ reducer: reducersSetup });
  const component = render(<Provider store={store}>{children}</Provider>);

  return { store, component };
};
