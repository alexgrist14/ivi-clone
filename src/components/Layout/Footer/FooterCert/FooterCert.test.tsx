import FooterCert from "./FooterCert";
import { renderWithProviders } from "/src/utils/test-utils";

describe("Footer cert", () => {
  test("should renders without errors", () => {
    const render = renderWithProviders(<FooterCert />);
    expect(render.component.container).toMatchSnapshot();
  });
});
