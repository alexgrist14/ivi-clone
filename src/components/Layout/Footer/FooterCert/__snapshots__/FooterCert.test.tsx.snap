// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer cert should renders without errors 1`] = `
<div>
  <div
    class="cert"
  >
    <a
      class="cert__link"
      href="https://www.ivi.ru/cert"
      target="_blank"
    >
      cert
    </a>
  </div>
</div>
`;
